eval.go

```diff
@@ -23,9 +23,68 @@ type EvalConfig struct {
 // semantic check on an AST tree. This will be called with the root node.
 type SemanticChecker func(ast.Node) error
 
+// EvalType represents the type of the output returned from a HIL
+// evaluation.
+type EvalType uint32
+
+const (
+	TypeInvalid EvalType = 0
+	TypeString  EvalType = 1 << iota
+	TypeList
+	TypeMap
+)
+
+//go:generate stringer -type=EvalType
```

Comment:
There was some reason we had to keep stringer generators in separate files in Terraform once upon a time, like so:

https://github.com/hashicorp/terraform/blob/master/helper/schema/valuetype.go

Is that no longer a valid constraint?